tests/test_objects.py

```diff
@@ -344,3 +344,85 @@ def test_external_api_aliases_failure(client, create_aliases_failure_patcher, da
 
     assert create_aliases_failure_patcher["data_upload_mock"].called
     assert create_aliases_failure_patcher["create_aliases_mock"].called
+
+
+@respx.mock
+def test_get_object_in_indexd(client):
+    """
+    Test the GET object endpoint when the provided key exists in indexd.
+    If the key is an indexd alias, the metadata returned should be
+    associated with the indexd GUID (did), not the alias itself.
+    If the key exists in indexd, the record should be returned regardless
+    of a 404 from MDS.
+    """
+    guid_or_alias = "dg.hello/test_guid"
+    indexd_did = "test_did"
+
+    # mock the request to indexd: GUID or alias found in indexd
+    indexd_url = f"{config.INDEXING_SERVICE_ENDPOINT}/{guid_or_alias}"
+    indexd_data = {"did": indexd_did, "size": 42}
+    indexd_mocked_request = respx.get(indexd_url, status_code=200, content=indexd_data)
+
+    # GET an object that exists in indexd but NOT in MDS
+    get_object_url = f"/objects/{guid_or_alias}"
+    resp = client.get(get_object_url)
+    assert indexd_mocked_request.called
+    assert resp.status_code == 200, resp.text
+    assert resp.json() == {"record": indexd_data, "metadata": {}}
+
+    # create metadata for this object
+    mds_data = dict(a=1, b=2)
+    client.post("/metadata/" + indexd_did, json=mds_data).raise_for_status()
+
+    # GET an object that exists in indexd AND in MDS
+    try:
+        resp = client.get(get_object_url)
+        assert indexd_mocked_request.called
+        assert resp.status_code == 200, resp.text
+        assert resp.json() == {"record": indexd_data, "metadata": mds_data}
+    finally:
+        client.delete("/metadata/" + indexd_did)
+
+
+@respx.mock
+def test_get_object_not_in_indexd(client):
```

Comment:
also possible to have an object in indexd but NOT in metadata db, we should test that too

Reply:
tested here https://github.com/uc-cdis/metadata-service/blob/b52ccf724feae18117717016d10111984fd7929d/tests/test_objects.py#L366

Reply:
:+1: ah, missed it, thanks